src/bsoncxx/include/bsoncxx/v_noabi/bsoncxx/stdx/string_view.hpp

```diff
@@ -14,101 +14,517 @@
+        if (pos > size()) {
+            throw std::out_of_range{"bsoncxx::stdx::basic_string_view::substr()"};
+        }
+        return basic_string_view(_begin + pos, (std::min)(count, size() - pos));
+    }
+
+    /**
+     * @brief Compare two strings lexicographically
+     *
+     * @param other The "right hand" operand of the comparison
+     * @returns `0` If *this == other
+     * @returns `n : n < 0` if *this is "less than" other.
+     * @returns `n : n > 0` if *this is "greater than" other.
+     */
+    constexpr int compare(basic_string_view other) const noexcept {
+        // Another level of indirection to support restricted C++11 constexpr
+        return _compare2(Traits::compare(data(), other.data(), (std::min)(size(), other.size())),
+                         other);
+    }
+
+    /**
+     * @brief Compare *this with the given C-string
+     *
+     * @returns compare(basic_string_view(cstr))
+     */
+    constexpr int compare(const_pointer cstr) const {
+        return compare(basic_string_view(cstr));
+    }
+
+    /**
+     * @brief Compare a substring of *this with `other`
+     *
+     * @returns substr(po1, count1).compare(other)
+     */
+    constexpr int compare(size_type pos1, size_type count1, basic_string_view other) const {
+        return substr(pos1, count1).compare(other);
+    }
+
+    /**
+     * @brief Compare a substring of *this with the given C-string
+     *
+     * @returns substr(pos1, count1, basic_string_view(cstr))
+     */
+    constexpr int compare(size_type pos1, size_type count1, const_pointer cstr) const {
+        return compare(pos1, count1, basic_string_view(cstr));
+    }
+
+    /**
+     * @brief Compare a substring of *this with a substring of `other`
+     *
+     * @returns substr(pos1, count1).compare(other.substr(pos2, count2))
+     */
+    constexpr int compare(size_type pos1,
+                          size_type count1,
+                          basic_string_view other,
+                          size_type pos2,
+                          size_type count2) const {
+        return substr(pos1, count1).compare(other.substr(pos2, count2));
+    }
+
+    /**
+     * @brief Compare a substring of *this with a string viewed through the given pointer+size
+     *
+     * @returns substr(pos1, count1).compare(basic_string_view(str, count2))
+     */
+    constexpr int compare(size_type pos1,
+                          size_type count1,
+                          const_pointer str,
+                          size_type count2) const {
+        return substr(pos1, count1).compare(basic_string_view(str, count2));
+    }
+
+    /**
+     * @brief Find the zero-based offset of the left-most occurrence of the given infix,
+     * starting with pos. If infix does not occur, returns npos.
+     */
+    bsoncxx_cxx14_constexpr size_type find(basic_string_view infix,
+                                           size_type pos = 0) const noexcept {
```

Comment:
```suggestion
    bsoncxx_cxx14_constexpr size_type find(basic_string_view infix, size_type pos = 0) const
        noexcept {
```
ClangFormat.

Reply:
Newer clang-format wants to break this line differently. Pain.